src/v2/view-builder/views/signin/SignInWithDeviceOption.js

```diff
@@ -73,5 +72,15 @@ export default View.extend({
     return {
       signInWithDeviceIsRequired: !!this.options.isRequired,
     };
+  },
+
+  postRender() {
+    if(this.options.isRequired) {
+      const app = this.options.appState.attributes.app;
```

Comment:
can we use optional here? appState won't be null but not sure if attributes could be null. something like `this.options.appState.attributes?.app`

Reply:
i was suggesting optional chaining in javascript https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining basically an easy for for NPE check

Reply:
cool done! Didn't knew this. Thanks!